src/database/tuple_desc.cpp

```diff
@@ -1,36 +1,113 @@
 #include "tuple_desc.h"
 
 namespace emerald {
-TupleDesc::TupleDesc TupleDesc::combine(TupleDesc td1, TupleDesc td2) {
-}
-
 TupleDesc::TupleDesc() {
 }
 
-TupleDesc::TupleDesc(Type type_array[]) {
+TupleDesc::TupleDesc(std::vector<Type> type_array) {
+  std::vector<std::string> empty_string = new std::vector<std::string>(0);
+  TupleDesc(type_array, empty_string);
 }
 
-TupleDesc::TupleDesc(Type type_array[], std::string field_array[]) {
+TupleDesc::TupleDesc(std::vector<Type> type_array, std::vector<std::string> field_array) {
+  types = new std::vector<Type>(type_array.size());
+  // memcpy ?
+  for (size_t i = 0; i < type_array.size(); i++) {
+      types.push_back(type_array[i]);
+  }
+  
+  names = new std::vector<std::string>(field_array.size());
+  for (size_t i = 0; i < field_array.size(); i++) {
+      names.push_back(field_array[i]);
+  }
 }
 
 TupleDesc::TupleDesc(const TupleDesc & other) {
 }
 
 int TupleDesc::get_number_fields() const {
+  return types.size();
 }
 
 std::string TupleDesc::get_field_name(int index) const {
+  if (index < 0 || i >= types.size()) {
+    // throw new NoSuchElementException("Index Out of Range."); 
+  } else {
+    return names[i];
+  }
 }
 
-int TupleDesc::get_index(std::string name) const {
+std::string TupleDesc::get_field_type(int index) const {
+  if (index < 0 || i >= types.size()) {
+    // throw new NoSuchElementException("Index Out of Range."); 
+  } else {
+    return types[i];
+  }
 }
 
-Type TupleDesc::get_type(int index) const {
+int TupleDesc::FieldNameToIndex(std::string name) {
+  if (name != NULL) {
+    for (size_t i = 0; i < name.length(); i++) {
+      if (name == names[i]) {
+        return i;
+      }
+    }
+  }
+  // throw new NoSuchElementException("Unknown Field Name.");
 }
 
 int TupleDesc::get_size() const {
+  int size = 0;
+  for (size_t i = 0; i < types.size(); i++) {
```

Comment:
Hmmm, is size_t needed here?

I think we can just use the size function here, since you changed the underlying implementation to vectors.

https://www.geeksforgeeks.org/vector-in-cpp-stl/